extractors/xtract_jsonxml.py

```diff
@@ -0,0 +1,248 @@
+                mh.update(by)
+
+        return mh
+
+    new_mdata = None
+
+    creds = event["creds"]
+    family_batch = event["family_batch"]
+
+    # This should be either 'local' or 'remote'.
+    extract_mode = event["extract_mode"]
+
+    downloader = GoogleDriveDownloader(auth_creds=creds)
+    assert extract_mode in ["remote", "local"], "Invalid extraction mode"
+
+    if extract_mode == "remote":
+        ta = time.time()
+        downloader.batch_fetch(family_batch=family_batch)
+        tb = time.time()
+
+        file_paths = downloader.success_files
+    elif extract_mode == "local":
+        ta = tb = 0  # Set both times to be zero so that transfer time is zero.
+        file_paths = []
+
+    if len(file_paths) == 0 and file_paths == "remote":
+        return {'family_batch': family_batch, 'error': True, 'tot_time': time.time() - t0,
+                'err_msg': "unable to download files"}
+
+    for family in family_batch.families:
+        img_path = family.files[0]['path']
+        # return img_path
+
+        new_mdata = xtract_jsonxml_main.extract_json_metadata(img_path)
+        new_mdata["min_hash"] = min_hash(img_path)
+        family.metadata = new_mdata
+
+    if extract_mode == "remote":
+        [os.remove(file_path) for file_path in downloader.success_files]
+
+    t1 = time.time()
+    # Return batch
+    return {'family_batch': family_batch, 'tot_time': t1 - t0, 'trans_time': tb - ta}
+
+
+from funcx.sdk.client import FuncXClient
+
+fxc = FuncXClient()
+
+location = '039706667969.dkr.ecr.us-east-1.amazonaws.com/xtract-images'
+description = 'JSON/XML Extractor'
+container_type = 'docker'
+name = "xtract/xtract-images"
+container_uuid = fxc.register_container(name, location, description, container_type)
+
+func = """
+def jsonxml_test(event):
+    import os
+    import tempfile
+    import time
+    import json
+    from shutil import copyfile, rmtree
+
+
+    from home_run.base import _get_file
+
+    # Make a temp dir and download the data
+    dir_name = tempfile.mkdtemp()
+    os.chdir(dir_name)
+
+    # A list of file paths
+    all_files = event['data']['inputs']
+
+    t0 = time.time()
+    mdata_list = []
+    for item in all_files:
+        ta = time.time()
+        dir_name = tempfile.mkdtemp()
+        file_id = item['file_id']
+        input_data = _get_file(item, dir_name)  # Download the file
+        tb = time.time()
+
+        # import json
+
+
+        t0 = time.time()
+        with open(input_data, 'r') as f:
+            data = json.load(f)
+
+        field_dict = {'keys': []}
+        for key in data:
+            field_dict['keys'].append(key)
+
+        t1 = time.time()
+
+        field_dict['extract_time'] = t1-t0
+
+        print(field_dict)
+
+        field_dict['file_id'] = file_id
+        field_dict['trans_time'] = tb-ta
+
+        mdata_list.append(field_dict)
+        rmtree(dir_name)
+
+
+    t1 = time.time()
+    return {'metadata': mdata_list, 'tot_time': t1-t0} 
+"""
+
+func2 = """
```

Comment:
You can delete the old 'free text' version of the funcX function. That's the format from nearly 2 year ago.